From 829316a224c4045190dc1a364b32cdb6eaa61d1f Mon Sep 17 00:00:00 2001
From: Francois Gaffie <francois.gaffie@intel.com>
Date: Tue, 30 Apr 2013 17:19:18 +0200
Subject: [PORT FROM MAIN] Add Stream Flags key to setParameters

BZ: 70126

Audio Route Manager needs to know the type of stream to take the
appropriate routing decision (eg direct Output, MP3 Offload).
It must be define as key and filled by the policy.

Feature: AudioComms
Category: Integration
Purpose: ABSP-Adapter
Origin: Internal
Upstream-candidate: yes
Change-Id: Ib0f538092d997254bab1af147eeca90647a88e5d
Orig-Change-Id: I87c8252acad1b3474f075cc1d83636920702d0b7
Signed-off-by: Francois Gaffie <francois.gaffie@intel.com>
---
 audio/AudioPolicyManagerBase.cpp |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index da2346c..dff9ecf 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -2438,6 +2438,7 @@ uint32_t AudioPolicyManagerBase::setOutputDevice(audio_io_handle_t output,
     ALOGV("setOutputDevice() changing device");
     // do the routing
     param.addInt(String8(AudioParameter::keyRouting), (int)device);
+    param.addInt(String8(AudioParameter::keyStreamFlags), (int)outputDesc->mFlags);
     mpClientInterface->setParameters(output, param.toString(), delayMs);
 
     // update stream volumes according to new device
-- 
1.7.4.1

